### PR TITLE
Feature/7471 service recipient put

### PR DESCRIPTION
**Pull request by SimonLLawrence**
STORY [AB#7412](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/7412)
TASK [AB#7471](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/7471)

### CODE REVIEW

File: NHSD.BuyingCatalogue.Ordering.Api.sln [hunk 1]
```diff
@@ -33,11 +33,6 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "NHSD.BuyingCatalouge.Orderi
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "NHSD.BuyingCatalogue.Ordering.Domain.UnitTests", "tests\NHSD.BuyingCatalogue.Ordering.Domain.UnitTests\NHSD.BuyingCatalogue.Ordering.Domain.UnitTests.csproj", "{B0EA4A17-4EE3-4557-842D-BDFA0136F5D3}"
 EndProject
-Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Solution Items", "Solution Items", "{0FA9669C-2656-4842-A2BB-C7ECE46B5CFD}"
```
Comment (sam-roberts-bjss) [inline NHSD.BuyingCatalogue.Ordering.Api.sln:36]: How come this has been removed?

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests.csproj [hunk 2]
```diff
@@ -14,6 +14,10 @@
     <None Remove="IdentitySigningKey.pfx" />
   </ItemGroup>
 
+  <ItemGroup>
```
Comment (sam-roberts-bjss) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests.csproj]: Is this needed?

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests.csproj [hunk 3]
```diff
@@ -45,4 +49,11 @@
     <ProjectReference Include="..\NHSD.BuyingCatalouge.Ordering.Api.Testing.Data\NHSD.BuyingCatalouge.Ordering.Api.Testing.Data.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
```
Comment (sam-roberts-bjss) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests.csproj]: Not sure this is need either. Please review.

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Builders/ServiceRecipientModelBuilder.cs [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+namespace NHSD.BuyingCatalogue.Ordering.Api.UnitTests.Controllers
+{
+    internal sealed class ServiceRecipientModelBuilder
+    {
+        private readonly ServiceRecipientModel _serviceRecipientModel;
+
+        private ServiceRecipientModelBuilder()
+        {
+            _serviceRecipientModel = new ServiceRecipientModel
```
Comment (sam-roberts-bjss) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Builders/ServiceRecipientModelBuilder.cs:11]: We agreed on Friday that the builders would use the slightly different style. So in this instance the `ServiceRecipientModel` object would be created in the `Build` method, not the constructor.

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Builders/ServiceRecipientsModelBuilder.cs [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+using NHSD.BuyingCatalogue.Ordering.Api.Models;
+using NUnit.Framework.Internal;
+
+namespace NHSD.BuyingCatalogue.Ordering.Api.UnitTests.Controllers
+{
+    internal sealed class ServiceRecipientsModelBuilder
+    {
+        private readonly ServiceRecipientsModel _serviceRecipientsModel;
+        private readonly List<ServiceRecipientModel> ServiceRecipientModelList = new List<ServiceRecipientModel>();
+        private ServiceRecipientsModelBuilder()
```
Comment (sam-roberts-bjss) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Builders/ServiceRecipientsModelBuilder.cs:10]: Spacing?

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Builders/ServiceRecipientsModelBuilder.cs [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+        {
+            _serviceRecipientsModel = new ServiceRecipientsModel()
+            {
+                ServiceRecipients = new List<ServiceRecipientModel>()
```
Comment (sam-roberts-bjss) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Builders/ServiceRecipientsModelBuilder.cs:14]: Same builder style, please apply previous comment here.

File: NHSD.BuyingCatalogue.Ordering.Api.sln (hunk 1 above)
Comment (SimonLLawrence) [reply NHSD.BuyingCatalogue.Ordering.Api.sln:36]: Not sure  not something i remember doing

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests.csproj (hunk 2 above)
Comment (SimonLLawrence) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests.csproj]: removed

File: NHSD.BuyingCatalogue.Ordering.Api.sln (hunk 1 above)
Comment (SimonLLawrence) [reply NHSD.BuyingCatalogue.Ordering.Api.sln:36]: Not sure what to do with this, I don't remember removing it, everything builds and passes. Doing a quick search on the web its only used for files  added to folder that are not part of a project.  Do we have any of these ?

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/ServiceRecipients Put.feature [hunk 7]
```diff
@@ -0,0 +1,120 @@
+﻿Feature: Update the Service Recipients for an order
```
Comment (sam-roberts-bjss) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/ServiceRecipients Put.feature:1]: Should a test be added to check the last updated by name and/or ID is updated when sending an update to the service recipients list?

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/AccessTokenSteps.cs [hunk 8]
```diff
@@ -61,5 +61,44 @@ public void TheUserIsLoggedInWithRoleForOrganisation(string role, string organis
             var token = builder.BuildToken();
             _context[ScenarioContextKeys.AccessToken] = token;
         }
+
+        [Given(@"the user (.*) with id (.*)is logged in with the (Buyer|Authority|Readonly-Buyer) role for organisation (.*)")]
```
Comment (sam-roberts-bjss) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/AccessTokenSteps.cs:65]: As mentioned on the call I don't think we need this extra step which is duplicating all the claims information that is already in another step. On the call we agreed to change the initial data in the database & use the details of the currently logged in user.

**Pull request merged**